frontend/src/app/libs/common-components/components/datepicker/datepicker.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Input, ChangeDetectionStrategy, Output, EventEmitter, OnInit, ElementRef, Optional, Host, SkipSelf, forwardRef } from '@angular/core';
+import { MAT_DATE_FORMATS, DateAdapter } from '@angular/material/core';
+import { AppDateAdapter, APP_DATE_FORMATS } from './dateadapter';
+import { ControlValueAccessor, FormControl, ControlContainer, FormGroup, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ComponentTheme } from '../../shared/component-theme.enum';
+
+@Component({
+	selector: 'app-datepicker',
+	templateUrl: './datepicker.component.html',
+	styleUrls: ['./datepicker.component.scss'],
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => DatepickerComponent),
+			multi: true
+		},
+		{ provide: DateAdapter, useClass: AppDateAdapter },
+		{ provide: MAT_DATE_FORMATS, useValue: APP_DATE_FORMATS }
+	],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DatepickerComponent implements ControlValueAccessor, OnInit {
+	@Input() public formControlName: string;
+
+	@Input() public label: string;
+
+	@Input() public value: number | string | boolean | null | undefined;
+
+	@Input() public formControl: FormControl;
+
+	@Input() public theme: ComponentTheme = ComponentTheme.Light;
+
+	@Output() public valueChange: EventEmitter<number | string | boolean | null | undefined> = new EventEmitter();
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef,
+	) { }
+
+	public ngOnInit(): void {
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
```

Comment:
Hm, this code like in Ckeditor component.
Is it possible to use polymorphism here?
Maybe we can extend this component from Ckeditor, isn't it?